tests/api2/test_certs.py

```diff
@@ -264,3 +264,40 @@ def test_revoking_ca():
             assert set(c['certificate'] for c in root_ca['revoked_certs']) == check_set, root_ca
         finally:
             call('certificate.delete', cert['id'], job=True)
+
+
+def test_certs_existence_on_filesystem():
+    with intermediate_certificate_authority('root_ca', 'intermediate_ca') as (root_ca, intermediate_ca):
+        with certificate_signing_request('csr_test') as csr:
+            with intermediate_certificate_authority('root_ca2', 'intermediate_ca2') as (root_ca2, intermediate_ca2):
+                # no-op
+                pass
+            with certificate_signing_request('csr_test2') as csr2:
+                pass
+
+            cert = call('certificate.create', {
+                'name': 'cert_test',
+                'signedby': intermediate_ca['id'],
+                'create_type': 'CERTIFICATE_CREATE_INTERNAL',
+                **get_cert_params(),
+            }, job=True)
+            try:
+                certs = call('certificate.query')
+                cas = call('certificateauthority.query')
+                expected_files = {'/etc/certificates/CA'}
+                for cert in certs + cas:
+                    if cert['chain_list']:
+                        expected_files.add(cert['certificate_path'])
+                    if cert['privatekey']:
+                        expected_files.add(cert['privatekey_path'])
+                    if cert['cert_type_CSR']:
+                        expected_files.add(cert['csr_path'])
+                    if any(cert[k] for k in ('CA_type_existing', 'CA_type_internal', 'CA_type_intermediate')):
+                        expected_files.add(cert['crl_path'])
+                found_files = {
+                    f['path']
+                    for p in ('/etc/certificates', '/etc/certificates/CA') for f in call('filesystem.listdir', p)
+                }
+                assert found_files == expected_files, f'Expected {expected_files!r} but found {found_files!r}'
```

Comment:
pytest assertion error will dump the set difference, no need to add custom message